### PR TITLE
update docs to promote PodReadyToStartContainersCondition into beta

### PR DESCRIPTION
**Pull request by charles-chenzz**
https://github.com/kubernetes/enhancements/issues/4138

### CODE REVIEW

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md
Comment (kannon92) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:173]:
```suggestion
| `PodReadyToStartContainersCondition` | `false` | Alpha | 1.28 | 1.29 |
| `PodReadyToStartContainersCondition` | `true` | Beta | 1.29 | |
```

Comment (kannon92) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:173]: Yes, thank you.

Comment (charles-chenzz) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:174]: updated

**Comment by harshitasao** [comment]
Hi @charles-chenzz  , Please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before **Tuesday 14th November 2023**.
Thank you!

**Comment by sftim** [comment]
Thanks

Please split the blog article into a separate PR that targets the _main_ branch.

/hold
pending that happening

**Review by sftim — CHANGES_REQUESTED** [review]
You'll also need to change https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/ I think

**Comment by charles-chenzz** [comment]
> You'll also need to change https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/ I think

I think it's better to change it when this feature promote to GA

**Comment by charles-chenzz** [comment]
> Please split the blog article into a separate PR that targets the _main_ branch.

done in #43932

**Comment by kannon92** [comment]
> > You'll also need to change https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/ I think
> 
> I think it's better to change it when this feature promote to GA

I disagree.  

https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-conditions should be updated to say that PodReadyToStartContainers is a beta-feature.  

We mention this in a few places and it'd be worth updating the docs to reference that this is a beta feature.

**Comment by sftim** [comment]
> > You'll also need to change https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/ I think
> 
> I think it's better to change it when this feature promote to GA

I'll explain; **this is me writing as a technical lead for SIG Docs**.

We'd expect docs for beta features to be pretty much at GA quality. A few snags are OK but missing a major update isn't, especially if the omission wasn't an accident.

Please make the change. `PodReadyToStartContainersCondition` (the associated feature gate) is enabled by default in the upcoming release.

**Comment by sftim** [comment]
/hold cancel

Blog changes are in their own PR.

**Comment by charles-chenzz** [comment]
> Please make the change. `PodReadyToStartContainersCondition` (the associated feature gate) is enabled by default in the upcoming release.

the changes should happen on which PR? the blog one or this one?

**Comment by sftim** [comment]
> the changes should happen on which PR? the blog one or this one?

_this_ PR

**Comment by sftim** [comment]
The `PodReadyToStartContainers` condition will appear by default in Kubernetes v1.29 clusters, right?

**Comment by kannon92** [comment]
> The `PodReadyToStartContainers` condition will appear by default in Kubernetes v1.29 clusters, right?

Correct.

**Comment by kannon92** [comment]
/hold

let’s address @sftim comments before we merge this.

**Comment by kannon92** [comment]
@charles-chenzz do you have some time Thursday or Friday to work on this? I think docs are due this week so it’s be nice to get this PR mergable

**Comment by charles-chenzz** [comment]
> @charles-chenzz do you have some time Thursday or Friday to work on this? I think docs are due this week so it’s be nice to get this PR mergable

I got time to work on this PR later today(16:00 in my timezone now, got some time in 20:00-22:00) and Friday. @kannon92

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/pod-lifecycle.md:167]:
```suggestion
* `PodReadyToStartContainers`: (beta feature; enabled by default) the
```

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/pod-lifecycle.md:248]:
```suggestion
During its early development, this condition was named `PodHasNetwork`.
```

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/pod-lifecycle.md:254]:
```suggestion
runtime sandbox and configure networking for the Pod. If the
`PodReadyToStartContainersCondition`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is enabled
(it is enabled by default for Kubernetes {{< skew currentVersion >}}), the
```

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/pod-lifecycle.md:255]:
```suggestion
kubelet reports whether a pod has reached this initialization milestone through
```

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/pod-lifecycle.md:252]:
```suggestion
After a Pod gets scheduled on a node, it needs to be admitted by the kubelet and
to have any required storage volumes mounted. Once these phases are complete,
the kubelet works with
```

Comment (charles-chenzz) [reply content/en/docs/concepts/workloads/pods/pod-lifecycle.md]: updated

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md
Comment (charles-chenzz) [reply content/en/docs/concepts/workloads/pods/pod-lifecycle.md]: updated

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md
Comment (charles-chenzz) [reply content/en/docs/concepts/workloads/pods/pod-lifecycle.md]: updated

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md
Comment (kannon92) [inline content/en/docs/concepts/workloads/pods/pod-lifecycle.md:243]: @sftim wdyt about changing the name of this section? This was the name of the old condition and we renamed it to PodReadyToStartContainers.

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md
Comment (kannon92) [inline content/en/docs/concepts/workloads/pods/pod-lifecycle.md:259]:
```suggestion
`PodReadyToStartContainers` condition will be added to the `status.conditions` field of a Pod.
```

**Comment by kannon92** [comment]
Looking much better @charles-chenzz! Just one nit and then it LGTM.

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md
Comment (charles-chenzz) [reply content/en/docs/concepts/workloads/pods/pod-lifecycle.md]: updated

**Comment by harshitasao** [comment]
Hi, @charles-chenzz  ! Kubernetes v1.29 Docs shadow here. J**ust a reminder that the deadline to have this PR reviewed and merged is Tuesday, 28 November**. Let me know if you need any help!

CCing KEP owners: @kannon92

**Comment by katcosgrove** [comment]
Hi, @charles-chenzz! The deadline to have this PR *merged* is **tomorrow.** What can I do to help get this across the line as soon as possible?

@sftim Do you feel that the changes you requested here have been addressed?

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md
Comment (sftim) [reply content/en/docs/concepts/workloads/pods/pod-lifecycle.md:243]: How about “Pod sandbox readiness”, then?

**Comment by sftim** [comment]
/hold cancel

(when adding a hold, it's a good idea to make it clear what conditions need to be met in order to unhold)

File: content/en/docs/concepts/workloads/pods/pod-lifecycle.md
Comment (sftim) [reply content/en/docs/concepts/workloads/pods/pod-lifecycle.md:243]: We can make that change after this PR lands.

**Review by sftim — COMMENTED** [review]
/lgtm

**Pull request merged**
